index/server/main.go

```diff
@@ -111,14 +114,31 @@ func main() {
 	}
 
 	// Before starting the server, push the devfile artifacts to the registry
+	// Build sample_index.json and stack_index.json given index.json
+	var sampleIndex []indexSchema.Schema
+	var stackIndex []indexSchema.Schema
 	for _, devfileIndex := range index {
+		if devfileIndex.Type == "sample" {
+			sampleIndex = append(sampleIndex, devfileIndex)
+		} else if devfileIndex.Type == "stack" {
+			stackIndex = append(stackIndex, devfileIndex)
+		}
+
 		if len(devfileIndex.Resources) != 0 {
 			err := pushStackToRegistry(devfileIndex)
 			if err != nil {
 				log.Fatal(err.Error())
 			}
 		}
 	}
+	err = indexLibrary.CreateIndexFile(sampleIndex, sampleIndexPath)
```

Comment:
Rather than re-generating parts of the index.json for both stacks and samples, why not just do something like the following for each endpoint:

1) Parse the index.json file from disk
2) If samples are requested, return only the entries samples. If stacks are requested, return those.